lib/plugins/aws/invokeLocal/index.test.js

```diff
@@ -507,4 +529,189 @@ describe('AwsInvokeLocal', () => {
+          stdout: new EventEmitter().on('data', () => {}),
+          stdin: {
+            write: writeChildStub,
+            end: endChildStub,
+          },
+          on: (key, callback) => callback(),
+        }),
+      });
+
+      // Remove Node.js internal "require cache" contents and re-require ./index.js
+      delete require.cache[require.resolve('./index')];
+      delete require.cache[require.resolve('child_process')];
+
+      const AwsInvokeLocalMocked = require('./index'); // eslint-disable-line global-require
+
+      serverless.setProvider('aws', new AwsProvider(serverless));
+      awsInvokeLocalMocked = new AwsInvokeLocalMocked(serverless, options);
+
+      awsInvokeLocalMocked.options = {
+        stage: 'dev',
+        function: 'first',
+        functionObj: {
+          handler: 'handler.hello',
+          name: 'hello',
+          timeout: 4,
+        },
+        data: {},
+      };
+    });
+
+    afterEach(() => {
+      delete require.cache[require.resolve('./index')];
+      delete require.cache[require.resolve('child_process')];
+    });
+
+    it('spawns java process with correct arguments', () =>
+      awsInvokeLocalMocked.callJavaBridge(
+        __dirname,
+        'com.serverless.Handler',
+        '{}'
+      ).then(() => {
+        expect(writeChildStub.calledOnce).to.be.equal(true);
+        expect(endChildStub.calledOnce).to.be.equal(true);
+        expect(writeChildStub.calledWithExactly('{}')).to.be.equal(true);
+      })
+    );
+  });
+
+  describe('#invokeLocalJava', () => {
+    let callJavaBridgeStub;
+
+    beforeEach(() => {
+      callJavaBridgeStub = sinon.stub(awsInvokeLocal, 'callJavaBridge').resolves();
+      awsInvokeLocal.options = {
+        stage: 'dev',
+        function: 'first',
+        functionObj: {
+          handler: 'handler.hello',
+          name: 'hello',
+          timeout: 4,
+        },
+        data: {},
+      };
+
+      serverless.cli = new CLI(serverless);
+      sinon.stub(serverless.cli, 'consoleLog');
+    });
+
+    afterEach(() => {
+      serverless.cli.consoleLog.restore();
+      callJavaBridgeStub.restore();
+    });
+
+    it('should invoke callJavaBridge when bridge is built', () => {
+      const bridgePath = path.join(__dirname, 'java', 'target');
+      fs.mkdir(bridgePath);
+
+      awsInvokeLocal.invokeLocalJava(
+        'java',
+        'com.serverless.Handler',
+        __dirname,
+        {}
+      ).then(() => {
+        expect(callJavaBridgeStub.calledOnce).to.be.equal(true);
+        expect(callJavaBridgeStub.calledWithExactly(
+          __dirname,
+          'com.serverless.Handler',
+          JSON.stringify({
+            event: {},
+            context: {
+              name: 'hello',
+              version: 'LATEST',
+              logGroupName: '/aws/lambda/hello',
+              timeout: 4,
+            },
+          })
+        )).to.be.equal(true);
+        fs.rmdirSync(bridgePath);
```

Comment:
Can we put this into an `afterEach()` to ensure that this is always removed even if the test fails?